sound/soc/sof/pm.c

```diff
@@ -124,9 +128,13 @@ static int sof_suspend(struct device *dev, bool runtime_suspend)
 	struct snd_sof_dev *sdev = dev_get_drvdata(dev);
 	int ret;
 
-	/* do nothing if dsp suspend callback is not set */
-	if (!sof_ops(sdev)->suspend)
-		return 0;
+	/* do nothing if dsp suspend callbacks are not set */
+	if (runtime_suspend)
+		if (!sof_ops(sdev)->runtime_suspend)
+			return 0;
+	else
+		if (!sof_ops(sdev)->suspend)
+			return 0;
```

Comment:
@ranj063  Do we need these checks here ? they are already checked in snd_sof_dsp_suspend and snd_sof_dsp_runtime_suspend